app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java

```diff
@@ -110,22 +124,30 @@ private void setupPetDetailsView() {
         mPetBreed.setText(mPet.getBreedFullName());
         mPetDescription.setText(mPet.getDescription());
         mShelterContactName.setText(mPet.getContactName());
-        mShelterContactPhone.setText(mPet.getContactPhone());
-        mShelterLocation.setText(mPet.getLocation());
-        mShelterEmail.setText(mPet.getContactEmail());
+  //      mShelterContactPhone.setText(mPet.getContactPhone());
+  //      mShelterLocation.setText(mPet.getLocation());
+        addUnderline(mShelterLocation,mPet.getLocation());
+        addUnderline(mShelterEmail, mPet.getContactEmail());
+        addUnderline(mShelterContactPhone,mPet.getContactPhone());
+//        mShelterEmail.setText(mPet.getContactEmail());
     }
 
+    private void addUnderline(TextView textView,String textString) {
+        SpannableString content = new SpannableString(textString);
+        content.setSpan(new UnderlineSpan(), 0, content.length(), 0);
+        textView.setText(content);
+    }
     // open in map
     public void onLocate() {
         String address = mPet.getLocation();
-        // Looks like the address is always a PO Box # which won't help us to map the location !
-//        if (mPet.getContactAdress2().length() > 0) {
-//            address = mPet.getContactAdress1();
-//            address += " " + mPet.getLocation();
-//        } else
-//        {
-//            address = mPet.getLocation();
-//        }
+        // Looks like the address1 is always a PO Box # which won't help us to map the location !
+        if (mPet.getContactAddress2() != null && mPet.getContactAddress2().length() > 0) {
```

Comment:
someone shared this tip with me recently:

```
TextUtils.isEmpty(someString)
```
Handles both null and an empty string.